returnn/tf/layers/basic.py

```diff
@@ -550,6 +550,9 @@ def __init__(self, activation, **kwargs):
     if self.output_before_activation:
       self.output.placeholder = self.output_before_activation.y
 
+    if activation == "abs" and self.output.dtype == "complex64":
+      self.output.dtype = "float32"
```

Comment:
This should not be part of `__init__` but rather in `get_out_data_from_opts`.

Reply:
Actually, it is already. So I don't understand, why is this needed?

Reply:
I.e. that means that we should change the base to `LayerBase`.

Reply:
I agree that we should do things the right and clean way, I just did not know better. Are you sure we should use the `LayerBase`? I updated my fix using the `_ConcatInputLayer` as the base, then concatination of input is still possible and we do not have to change the existing logic in the `ActivationLayer's` `__init__` method.

Reply:
Ah right, `_ConcatInputLayer` makes more sense then.